weasyl/controllers/moderation.py

```diff
@@ -2,246 +2,205 @@
-        ])
-
-    POST = GET
-
-
-class modcontrol_closereport_(controller_base):
-    login_required = True
-    moderator_only = True
-
-    @define.token_checked
-    def POST(self):
-        form = web.input(reportid='', action='')
-        report.close(self.user_id, form)
-        raise web.seeother("/modcontrol/report?reportid=%s" % (form.reportid,))
-
-
-class modcontrol_contentbyuser_(controller_base):
-    login_required = True
-    moderator_only = True
-
-    def GET(self):
-        form = web.input(name='', features=[])
-
-        submissions = moderation.submissionsbyuser(self.user_id, form) if 's' in form.features else []
-        characters = moderation.charactersbyuser(self.user_id, form) if 'c' in form.features else []
-        journals = moderation.journalsbyuser(self.user_id, form) if 'j' in form.features else []
-
-        return define.webpage(self.user_id, "modcontrol/contentbyuser.html", [
-            form.name,
-            sorted(submissions + characters + journals, key=lambda item: item['unixtime'], reverse=True),
-        ])
-
-
-class modcontrol_massaction_(controller_base):
-    login_required = True
-    moderator_only = True
-
-    def POST(self):
-        form = web.input(action='', name='', submissions=[], characters=[], journals=[])
-        if form.action.startswith("zap-"):
-            # "Zapping" cover art or thumbnails is not a bulk edit.
-            if not form.submissions:
-                raise WeasylError("Unexpected")
-            submitid = int(form.submissions[0])
-            type = form.action.split("zap-")[1]
-            if type == "cover":
-                moderation.removecoverart(self.user_id, submitid)
-            elif type == "thumb":
-                moderation.removethumbnail(self.user_id, submitid)
-            elif type == "both":
-                moderation.removecoverart(self.user_id, submitid)
-                moderation.removethumbnail(self.user_id, submitid)
-            else:
-                raise WeasylError("Unexpected")
-            raise web.seeother("/submission/%i" % (submitid,))
+@moderator_only
+def modcontrol_(request):
+    return Response(define.webpage(request.userid, "modcontrol/modcontrol.html"))
+
+
+@moderator_only
+def modcontrol_finduser_(request):
+    return Response(define.webpage(request.userid, "modcontrol/finduser.html"))
+
+
+@moderator_only
+@token_checked
+def modcontrol_finduser_post_(request):
+    form = request.web_input(userid="", username="", email="")
+
+    return Response(define.webpage(request.userid, "modcontrol/finduser.html", [
+        # Search results
+        moderation.finduser(request.userid, form)
+    ]))
+
+
+@moderator_only
+def modcontrol_suspenduser_get_(request):
+    return Response(define.webpage(request.userid, "modcontrol/suspenduser.html",
+                                   [moderation.BAN_TEMPLATES, json.dumps(moderation.BAN_TEMPLATES)]))
+
+
+@moderator_only
+@token_checked
+def modcontrol_suspenduser_(request):
```

Comment:
nit: I've done a style of appending post_ to the post methods when there's also a get method. But since there's only two, it's no big deal. Should we change the other instances to this style of no `post_` on urls that are the 'default' behavior?